sonar-kotlin-plugin/src/test/java/org/sonarsource/kotlin/plugin/ContentHashCacheTest.kt

```diff
@@ -0,0 +1,110 @@
+import org.sonar.api.batch.sensor.cache.WriteCache
+import org.sonar.api.utils.log.LoggerLevel
+import org.sonarsource.kotlin.DummyInputFile
+import org.sonarsource.kotlin.DummyReadCache
+import org.sonarsource.kotlin.DummyWriteCache
+import org.sonarsource.kotlin.plugin.caching.ContentHashCache
+import org.sonarsource.kotlin.testing.AbstractSensorTest
+import java.nio.file.Path
+import java.security.MessageDigest
+
+class ContentHashCacheTest : AbstractSensorTest() {
+
+    private val pathToFile = Path.of("src", "test", "resources", "caching", "DummyFile.kt")
+    private val pathToFileChanged = Path.of("src", "test", "resources", "caching", "DummyFileChanged.kt")
+    private val dummyFile = DummyInputFile(pathToFile)
+    private val dummyFileChanged = DummyInputFile(pathToFileChanged)
+    private val hashAlgorithm = "MD5"
+    private val messageDigest = MessageDigest.getInstance(hashAlgorithm)
+    private val fileHash = messageDigest.digest(dummyFile.contents().byteInputStream().readAllBytes())
+
+    @Test
+    fun `test stored contents are present`() {
+        val contentHashCache = contentHashCacheOf("kotlin:contentHash:MD5:DummyFile.kt", fileHash)
+        assertThat(contentHashCache?.hasDifferentContentCached(dummyFile)).isFalse
+    }
+
+    @Test
+    fun `test non-stored contents are not present`() {
+        val contentHashCache = emptyContentHashCache()
+        assertThat(contentHashCache?.hasDifferentContentCached(dummyFile)).isTrue
+    }
+
+    @Test
+    fun `test changed file detection`() {
+        val key = "kotlin:contentHash:MD5:DummyFileChanged.kt"
+        val contentHashCache = contentHashCacheOf(key, fileHash)
+        assertThat(contentHashCache?.hasDifferentContentCached(dummyFileChanged)).isTrue
+    }
+
+    @Test
+    fun `test same key cannot be stored twice`() {
+        val readCache = DummyReadCache(mapOf())
+        val writeCache = DummyWriteCache()
+        writeCache.write("kotlin:contentHash:MD5:DummyFile.kt", fileHash)
+        val contentHashCache = cacheFromContextData(readCache, writeCache, true)
+        contentHashCache?.hasDifferentContentCached(dummyFile)
+        assertThat(logTester.logs(LoggerLevel.WARN)).contains("Cache already contains key kotlin:contentHash:MD5:DummyFile.kt")
+    }
+
+    @Test
+    fun `contentHashCache state is consistent with sensor context`() {
+        val contentHashCacheEnabled = emptyContentHashCache()
+        assertThat(contentHashCacheEnabled != null).isTrue
+        context.isCacheEnabled = false
+        val contentHashCacheDisabled = ContentHashCache.of(context)
+        assertThat(contentHashCacheDisabled == null).isTrue
+    }
+
+    @Test
+    fun `test NullPointerException on context with cache disabled`() {
+        context.isCacheEnabled = false
+        val contentHashCacheDisabled = ContentHashCache.of(context)
+        assertThrows<NullPointerException> { contentHashCacheDisabled!!.hasDifferentContentCached(dummyFile) }
+    }
```

Comment:
I think this test is redundant with the one above since the goal is to test the behavior of `ContentHashCache.of`